lib/nexpose/scheduled_maintenance.rb

```diff
@@ -0,0 +1,91 @@
+module Nexpose
+# Constants useful across the Nexpose module.
+# Configuration structure for scheduled maintenance.
+  class ScheduledMaintenance < APIObject
+    require 'json'
+    include JsonSerializer
+
+    # Whether or not this maintenance schedule is enabled. Defaults to true if not set
+    attr_accessor :enabled
+    # Valid schedule types: daily, hourly, monthly-date, monthly-day, weekly.
+    attr_accessor :schedule_type
+    # The repeat interval based upon type.
+    attr_accessor :schedule_interval
+    # The earliest date to generate the report on (in ISO 8601 format).
+    attr_accessor :schedule_start
+    # Whether the reindex task should run. Defaults to true if not set
+    attr_accessor :reindex
+    # Whether the compression task should run. Defaults to true if not set
+    attr_accessor :compress
+    # Whether the cleanup task should run. Defaults to true if not set
+    attr_accessor :cleanup
+    # Whether the maintenance should pause all local scans or wait for local scans to complete. Defaults to true if not set
+    attr_accessor :pause_local_scans
+    # Cancellation window in minutes for scheduled DB maintenance. Defaults to 0 if not set
+    attr_accessor :cancellation_period
+
+    def initialize(start:, enabled: true, type:, interval:, reindex: true, compress: true, cleanup: true, pause_local_scans: true, cancellation_period: 0)
```

Comment:
I know I said to make all of these default to true, but in hindsight the `reindex` option should actually be `false` by default. The time it takes vs what it accomplishes does not provide a high value in most situations.